File: exercises/reverse-a-string/reverse-string.js
/**
 * Reverse a string
 */

const reverseString = (str) => {
    // If str is not a String, return false
    if(typeof str !== 'string'){
        return false;
    }

    // If string length < 2, return string
    if(str.length < 2) {
        return str;
    }

    const strAsArray = str.split('');
    let reversed = '';
    for(let i = strAsArray.length -1; i >= 0; i--) {
        reversed+= strAsArray[i];
    }
    return reversed;
}

const quote = 'The quick brown fox jumped over the lazy dog.';

console.log(reverseString(quote));
console.log(reverseString('a'));
console.log(reverseString({ foo: 'bar' }));

File: exercises/reverse-a-string/reverse-string.py
# reverse a string

from ast import Str
from tokenize import String


def reverse_string(string):
    # if input not a string
    if isinstance(string, str) == False:
        return False

    # if string is empty or length = 1, return string
    length = len(string)

    if length < 2:
        return string
    
    counter = length -1
    reversed = ""
    while counter >= 0:
        reversed+= string[counter]
        counter-= 1
    return reversed

quote = "The quick brown fox jumped over the lazy dog."

print(reverse_string(quote))
print(reverse_string(True))